modules/3d/test/test_octree.cpp

```diff
@@ -0,0 +1,113 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+#include "test_precomp.hpp"
+
+// I don't know why Octree can't be called directly, I can only test the interface in this way.
+#if 1
+#include "../src/octree.cpp"
+#endif
+
+namespace opencv_test { namespace {
+
+using namespace cv;
+
+class OctreeTest: public::testing::Test
+{
+public:
+    std::vector<Point3f> pointcloud;
+    int pointCloudSize;
+    Point3f restPoint;
+    Octree treeTest;
+
+protected:
+    void SetUp() override
+    {
+        pointCloudSize = 1000;
+        maxDepth = 4;
+        pmin = Point3f(-1.0f, -1.0f, -1.0f);
+        pmax = Point3f(1.0f, 1.0f, 1.0f);
+
+        RNG rng_Point(12345); // set random seed for fixing output 3D point.
```

Comment:
OpenCV testing infrastructure should "fix" `theRNG()` state.

Reply:
I suggest to keep it this way, it looks quite natural, safe and efficient.

Reply:
@zihaomu, please, remove `rng_Point` from the class.
Before the point cloud initialization loop simply put

```
RNG& rng_Point = theRNG();
```